### DISCUSSION TITLE
HazelCast client is not working with GenericContainer

### DISCUSSION THREAD

**Discussion by haider665** [testcontainers/testcontainers-java#8051] [Q&A]
I have a hazelcast-client yaml and just a single component. I have tried using the example mentioned in this test container github [example](https://github.com/testcontainers/testcontainers-java/blob/main/examples/hazelcast/src/test/java/org/testcontainers/examples/HazelcastTest.java). But the application is still waiting for hazercast to run locally.

hazelcast-client.yaml

```
hazelcast-client:
  network:
    cluster-members:
      - 127.0.0.1
    smart-routing: true
    redo-operation: true
```

HazelCastDemo.java

```
@Component
public class HazelcastDemo {

    private String cacheMap = "test-map";

    @Autowired
    private HazelcastInstance hazelcastInstance;

    public void createDummyMap() {
        IMap<String, Integer> cachedMap = hazelcastInstance.getMap(cacheMap);
        cachedMap.put("1", 1);
    }
}
```
HazelcastClientApplicationTests.java

```
@SpringBootTest
class HazelcastClientApplicationTests {

    private String cacheMap = "test-map";

    @Autowired
    private HazelcastDemo hazelcastDemo;

    @Autowired
    private HazelcastInstance hazelcastInstance;

    static GenericContainer<?> container = 
            new GenericContainer<>(DockerImageName.parse("hazelcast/hazelcast:latest"))
            .withExposedPorts(5701);

    @BeforeAll
    public static void singleHazelcastContainer() {
        container.start();

        ClientConfig clientConfig = new ClientConfig();
        clientConfig
                .getNetworkConfig()
                .addAddress(container.getHost() + ":" + container.getFirstMappedPort());
    }

    @BeforeEach
    void cleanUp() {
        hazelcastInstance.getMap(cacheMap).clear();
    }

    @Test
    void contextLoads() {
        hazelcastDemo.createDummyMap();

        assertFalse(hazelcastInstance.getMap(cacheMap).isEmpty());
    }

}
```

**Comment by eddumelendez** [comment]
If you want us to spend some time looking at it, please share a project that reproduces the issue.

**Reply by haider665** [reply]
here is the link of that project. HazelcastClientApplicationTests.cacheInitTest() produced that issue.

https://github.com/haider665/hazelcast-client

**Reply by eddumelendez** [reply]
Sorry but I think this is not a Testcontainers question but rather about how to configure your Spring Boot project. In your test, you create an instance of `clientConfig` but it is not used at all by `hazelcastInstance`.

**Reply by haider665** [reply]
This application is reading the hazelcast client config from `hazelcast-client.yaml`. If there is no hazelcast instance running locally then this application failed to start. 

Is there any way to use hazelcast generic container and this `clientConfig` so that `HazelcastDemo.hazelcastInstance` uses that container instead of `hazelcast-client.yaml`?
